official/vision/beta/projects/yolo/ops/preprocessing_ops.py

```diff
@@ -0,0 +1,434 @@
+  """
+  with tf.name_scope('translate_boxs'):
+    x = box[..., 0] + translate_x
+    y = box[..., 1] + translate_y
+    box = tf.stack([x, y, box[..., 2], box[..., 3]], axis=-1)
+    box.set_shape([None, 4])
+  return box
+
+
+def translate_image(image, translate_x, translate_y):
+  """Randomly translate the image.
+        Args:
+            image: a `Tensor` representing the image.
+            translate_x: a `Tensor` represting the translation on the x-axis.
+            translate_y: a `Tensor` represting the translation on the y-axis.
+        Returns:
+            box: a `Tensor` representing the augmented boxes.
+        """
+  with tf.name_scope('translate_image'):
+    if (translate_x != 0 and translate_y != 0):
+      image_jitter = tf.convert_to_tensor([translate_x, translate_y])
+      image_jitter.set_shape([2])
+      image = tfa.image.translate(
+          image, image_jitter * tf.cast(tf.shape(image)[1], tf.float32))
+  return image
+
+
+def pad_max_instances(value, instances, pad_value=0, pad_axis=0):
+  shape = tf.shape(value)
+  dim1 = shape[pad_axis]
+  take = tf.math.reduce_min([instances, dim1])
+  value, _ = tf.split(
+      value, [take, -1], axis=pad_axis)  # value[:instances, ...]
+  pad = tf.convert_to_tensor([tf.math.reduce_max([instances - dim1, 0])])
+  nshape = tf.concat([shape[:pad_axis], pad, shape[(pad_axis + 1):]], axis=0)
+  pad_tensor = tf.fill(nshape, tf.cast(pad_value, dtype=value.dtype))
+  value = tf.concat([value, pad_tensor], axis=pad_axis)
+  return value
+
+
+def fit_preserve_aspect_ratio(image,
+                              boxes,
+                              width=None,
+                              height=None,
+                              target_dim=None):
+  if width is None or height is None:
+    shape = tf.shape(image)
+    if tf.shape(shape)[0] == 4:
+      width = shape[1]
+      height = shape[2]
+    else:
+      width = shape[0]
+      height = shape[1]
+
+  clipper = tf.math.maximum(width, height)
+  if target_dim is None:
+    target_dim = clipper
+
+  pad_width = clipper - width
+  pad_height = clipper - height
+  image = tf.image.pad_to_bounding_box(image, pad_width // 2, pad_height // 2,
+                                       clipper, clipper)
+
+  boxes = box_ops.yxyx_to_xcycwh(boxes)
+  x, y, w, h = tf.split(boxes, 4, axis=-1)
+
+  y *= tf.cast(width / clipper, tf.float32)
+  x *= tf.cast(height / clipper, tf.float32)
+
+  y += tf.cast((pad_width / clipper) / 2, tf.float32)
+  x += tf.cast((pad_height / clipper) / 2, tf.float32)
+
+  h *= tf.cast(width / clipper, tf.float32)
+  w *= tf.cast(height / clipper, tf.float32)
+
+  boxes = tf.concat([x, y, w, h], axis=-1)
+
+  boxes = box_ops.xcycwh_to_yxyx(boxes)
+  image = tf.image.resize(image, (target_dim, target_dim))
+  return image, boxes
+
+
+def get_best_anchor(y_true, anchors, width=1, height=1):
+  """
+    get the correct anchor that is assoiciated with each box using IOU betwenn
```

Comment:
typo: between.

Reply:
Fixed in https://github.com/tensorflow/models/pull/9493/commits/28559cab79d1ed7521a803844e096c5e79064bef